### ISSUE TITLE
Existing schema attributes cannot be modified or deleted.

### ISSUE THREAD

**Issue by tscole** [aws-amplify/amplify-cli#13095]
### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v18.16.0

### Amplify CLI Version

12.2.3

### What operating system are you using?

Mac

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

Custom attributes added in yml for cognition user pool:

```yaml
      Schema:
        - Name: email
          Required: true
          Mutable: true
        - Name: stripe_id
          Mutable: 'true'
          AttributeDataType: String
          StringAttributeConstraints:
            MaxLength: '256'
            MinLength: '1'
        - Name: stripe_sub_id
          Mutable: 'true'
          AttributeDataType: String
          StringAttributeConstraints:
            MaxLength: '256'
            MinLength: '1'
        - Name: stripe_plan_name
          Mutable: 'true'
          AttributeDataType: String
        - Name: company
          Mutable: 'true'
          AttributeDataType: String
        - Name: vat_number
          Mutable: 'true'
          AttributeDataType: String
        - Name: address_1
          Mutable: 'true'
          AttributeDataType: String
        - Name: address_2
          Mutable: 'true'
          AttributeDataType: String
        - Name: city
          Mutable: 'true'
          AttributeDataType: String
        - Name: postcode
          Mutable: 'true'
          AttributeDataType: String
        - Name: state
          Mutable: 'true'
          AttributeDataType: String
        - Name: country
          Mutable: 'true'
          AttributeDataType: String
```

### Describe the bug

This is an old project that hasn't had any work for a while.
After upgrading to amplify cli v12.2.3 the .yml file is deleted and override.ts is required for custom attributes.

I have the following override.ts in place:

```ts
import { AmplifyAuthCognitoStackTemplate } from "@aws-amplify/cli-extensibility-helper";

export function override(resources: AmplifyAuthCognitoStackTemplate) {
  resources.userPool.schema = [
    ...(resources.userPool.schema as any[]), // Carry over existing attributes (example: email)
    {
      attributeDataType: "String",
      mutable: true,
      name: "email",
      required: true,
    },
    {
      attributeDataType: "String",
      mutable: true,
      name: "stripe_id",
      stringAttributeConstraints: {
        maxLength: "256",
        minLength: "1",
      },
    },
    {
      attributeDataType: "String",
      mutable: true,
      name: "stripe_sub_id",

      stringAttributeConstraints: {
        maxLength: "256",
        minLength: "1",
      },
    },
    {
      attributeDataType: "String",
      mutable: true,
      name: "stripe_plan_name",
    },
    {
      attributeDataType: "String",
      mutable: true,
      name: "company",
    },
    {
      attributeDataType: "String",
      mutable: true,
      name: "address_1",
    },
    {
      attributeDataType: "String",
      mutable: true,
      name: "address_2",
    },
    {
      attributeDataType: "String",
      mutable: true,
      name: "city",
    },
    {
      attributeDataType: "String",
      mutable: true,
      name: "state",
    },
    {
      attributeDataType: "String",
      mutable: true,
      name: "postcode",
    },
    {
      attributeDataType: "String",
      mutable: true,
      name: "country",
    },
    {
      attributeDataType: "String",
      mutable: true,
      name: "vat_number",
    },
  ];
}
```

However attempting amplify push results in the error  **Existing schema attributes cannot be modified or deleted.**

I am therefore unable to do any further work to the back end of this project as all pushes fail.

### Expected behavior

Pushing should work because override.ts should mean that cognito attributes aren't being changed.

### Reproduction steps

You will need to start with an earlier version of the amplify cli and use the supplied yaml file to create the custom attributes. Upgrade to the latest CLI and then attempt a push using the supplied override.ts. 

### Project Identifier

dbbb418abb38539d3a09d33eacc39132

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Before submitting, please confirm:

- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

**Comment by ykethan** [comment]
Hey @tscole, thank you for reaching out. if you add a `console.log(resources.userPool.schema)` at the top, do you observe the `email` attribute bring printed out in the terminal? if yes, could you remove the following block and retry the push. 
`{
      attributeDataType: "String",
      mutable: true,
      name: "email",
      required: true,
    },
  `

additionally, were there any pushes before this that was successful and attributes were added via the override?

**Comment by tscole** [comment]
Hi @ykethan Where exactly do you do you want me to add the `console.log` ? I tried adding it just above `resources.userPool.schema = [` but nothing at all gets output to the terminal.

**Comment by tscole** [comment]
No attributes have been added via override. They were all added by the original yaml file.

**Comment by ykethan** [comment]
apologies, yes add the console log right under the override handler. could you run `amplify build` and let us know if you observe an output.
similar to
![image](https://github.com/aws-amplify/amplify-cli/assets/87995712/c4fdfb9e-bc2a-4773-bd19-a472f4efc2ad)

**Comment by tscole** [comment]
Nothing at all happens if I run `amplify build`. That's not a command I usually use so I don't know if thats a new issue? Running `amplify push` doesn't output anything like your screenshot above.

**Comment by tscole** [comment]
Running `amplify build --debug` gives the following output and nothing else:

`Overrides functionality is not implemented for this category`

**Comment by ykethan** [comment]
thank you for hopping on the call, I was great chatting with you.

We were able to mitigate the issue by updating the Amplify cli to v7.6.15. Then `amplify auth override` to add the schema attributes.
As the version utilized `nodej12` in the auth lambda functions we updated the Amplify CLI version to v8.2.0. where we an observe error
```
Command was killed with SIGSEGV (Segmentation fault): yarn install
```
To mitigate this error message we cd'd into the backend folder and did a `npm install`. Additionally, cd'd into the backend `/function/<function-name>/src` and installed using npm.
The push succeeded when running `amplify push`
